src/credentials/providers/envVarsCredentialsProvider.ts

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Credentials, EnvironmentCredentials } from "aws-sdk";
+import { EnvironmentVariables } from "../../shared/environmentVariables";
+import { CredentialType } from "../../shared/telemetry/telemetry.gen";
+import { CredentialsId, CredentialsProviderType } from './credentials';
+import { EnvironmentCredentialsProvider } from "./environmentCredentialsProvider";
+
+/**
+ * Credentials given by environment variables.
+ *
+ * @see CredentialsProviderType
+ */
+export class EnvVarsCredentialsProvider implements EnvironmentCredentialsProvider {
+    public static readonly AWS_ENV_VAR_PREFIX: string = 'AWS'
+
+    private credentials: EnvironmentCredentials | undefined
+
+    public async isAvailable(): Promise<boolean> {
+        const env = process.env as EnvironmentVariables
+        return env.AWS_ACCESS_KEY_ID && env.AWS_SECRET_ACCESS_KEY ? true : false
+    }
+
+    public getCredentialsId(): CredentialsId {
+        return {
+            credentialSource: this.getProviderType(),
+            credentialTypeId: 'variables',
+        }
+    }
+
+    public static getProviderType(): CredentialsProviderType {
+        return 'env'
+    }
+
+    public getProviderType(): CredentialsProviderType {
+        return EnvVarsCredentialsProvider.getProviderType()
+    }
+
+    public getTelemetryType(): CredentialType {
+        return 'other' // TODO: backfill 'envVars' back into telemetry type
```

Comment:
since https://github.com/aws/aws-toolkit-common/commit/630c6de54da5fde6df3305b26b905875b24bf45f , envvars is classified as a source instead of a "material type".